mlflow/store/artifact/cloud_artifact_repo.py

```diff
@@ -225,6 +220,8 @@ def _parallelized_download_from_cloud(self, file_size, remote_file_path, local_p
 
                 for i in failed_downloads:
                     download_chunk(i, _DOWNLOAD_CHUNK_SIZE, new_headers, local_path, new_signed_uri)
+        except MlflowException:
+            raise
```

Comment:
Why do we need this?

Reply:
To avoid rewrapping `MlflowException`

Reply:
If MlflowException happens, should we remove local_path as well?

Reply:
Yeah, we should. Good catch!